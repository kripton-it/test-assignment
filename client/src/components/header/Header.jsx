import React from 'react';

import * as Styles from './styles';

const Header = ({ location }) => {
  const { pathname } = location;
  const logo = pathname === "/"
    ? (
      <Styles.Logo>
        Read . us
      </Styles.Logo>
    ) : (
      <Styles.Logo>
        <Styles.LogoLink to="/">
          Read . us
        </Styles.LogoLink>
      </Styles.Logo>
    );
  const menu = (
    <Styles.MenuList>
      <Styles.MenuItem>
        <Styles.MenuLink href="#">
          New
            </Styles.MenuLink>
      </Styles.MenuItem>
      <Styles.MenuItem>
        <Styles.MenuLink href="#">
          Non-fiction
            </Styles.MenuLink>
      </Styles.MenuItem>
      <Styles.MenuItem>
        <Styles.MenuLink href="#">
          Children
            </Styles.MenuLink>
      </Styles.MenuItem>
      <Styles.MenuItem>
        <Styles.MenuLink href="#">
          Fiction
            </Styles.MenuLink>
      </Styles.MenuItem>
      <Styles.MenuItem>
        <Styles.MenuLink href="#">
          Education
            </Styles.MenuLink>
      </Styles.MenuItem>
    </Styles.MenuList>
  );
  return (
    <Styles.HeaderContainer>
      {logo}
      <Styles.Navigation>
        {menu}
      </Styles.Navigation>
    </Styles.HeaderContainer>
  );
};

export default Header;