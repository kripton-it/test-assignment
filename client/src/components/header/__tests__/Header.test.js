import React from "react";
import { shallow } from "enzyme";
import toJson from 'enzyme-to-json';

import Header from "../Header";

const props1 = {
  location: {
    pathname: '/'
  }
};

const props2 = {
  location: {
    pathname: '/qwerty'
  }
};

describe("Header component:", () => {
  it("expect to render with path === '/'", () => {
    const wrapper = shallow(<Header {...props1} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("expect to render with path !== '/'", () => {
    const wrapper = shallow(<Header {...props2} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
