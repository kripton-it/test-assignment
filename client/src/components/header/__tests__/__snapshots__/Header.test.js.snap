// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component: expect to render with path !== '/' 1`] = `
<HeaderContainer>
  <Logo>
    <LogoLink
      to="/"
    >
      Read . us
    </LogoLink>
  </Logo>
  <Navigation>
    <MenuList>
      <MenuItem>
        <MenuLink
          href="#"
        >
          New
        </MenuLink>
      </MenuItem>
      <MenuItem>
        <MenuLink
          href="#"
        >
          Non-fiction
        </MenuLink>
      </MenuItem>
      <MenuItem>
        <MenuLink
          href="#"
        >
          Children
        </MenuLink>
      </MenuItem>
      <MenuItem>
        <MenuLink
          href="#"
        >
          Fiction
        </MenuLink>
      </MenuItem>
      <MenuItem>
        <MenuLink
          href="#"
        >
          Education
        </MenuLink>
      </MenuItem>
    </MenuList>
  </Navigation>
</HeaderContainer>
`;

exports[`Header component: expect to render with path === '/' 1`] = `
<HeaderContainer>
  <Logo>
    Read . us
  </Logo>
  <Navigation>
    <MenuList>
      <MenuItem>
        <MenuLink
          href="#"
        >
          New
        </MenuLink>
      </MenuItem>
      <MenuItem>
        <MenuLink
          href="#"
        >
          Non-fiction
        </MenuLink>
      </MenuItem>
      <MenuItem>
        <MenuLink
          href="#"
        >
          Children
        </MenuLink>
      </MenuItem>
      <MenuItem>
        <MenuLink
          href="#"
        >
          Fiction
        </MenuLink>
      </MenuItem>
      <MenuItem>
        <MenuLink
          href="#"
        >
          Education
        </MenuLink>
      </MenuItem>
    </MenuList>
  </Navigation>
</HeaderContainer>
`;
